server/src/infra/repositories/smart-info.repository.ts

```diff
@@ -52,12 +52,18 @@ export class SmartInfoRepository implements ISmartInfoRepository {
     await this.assetRepository.manager.transaction(async (manager) => {
       await manager.query(`SET LOCAL vectors.k = '${numResults}'`);
       await manager.query(`SET LOCAL vectors.enable_prefilter = on`);
-      results = await manager
+      let query = manager
         .createQueryBuilder(AssetEntity, 'a')
         .innerJoin('a.smartSearch', 's')
         .where('a.ownerId IN (:...userIds )')
-        .andWhere('a.isVisible = true')
-        .andWhere('a.isArchived = false')
+        .andWhere('a.isVisible = true');
+
+      if (!searchArchived) {
+        query = query.andWhere('a.isArchived = false');
+      }
+
+      results = await query
+        .andWhere('a.isArchived = true')
```

Comment:
Won't you have `andWhere('a.isArchived = false')` and `andWhere('a.isArchived = true')` both in the query if `searchArchived` is false?

Reply:
You are absolutely correct, and it's embarrassing that I missed that. I'm not sure how my testing archived vs. non-archived search results was returning properly with that conditional. I'm fixing and will push a new build. I've been rebasing to create a single commit, is that the preferred process you follow for PRs for Immich?

Reply:
I've pushed a fix for this. I also repeated my testing, and confirmed the search is working as expected both with and without the query parameter.

Reply:
It's not embarrassing at all! I miss such things all the time. It happens when you're in the flow. That's what reviews are for; if everyone would make no typos/mistakes ever, we wouldn't need reviews either :D